tests/test_interconnect/test_interconnect_cgra.py

```diff
@@ -405,3 +406,135 @@ def test_interconnect_sram(cw_files, add_pd, io_sides):
                                magma_output="coreir-verilog",
                                directory=tempdir,
                                flags=["-Wno-fatal"])
+
+
+@pytest.mark.parametrize("add_pd", [True, False])
+@pytest.mark.parametrize("depth", [1, 10, 100])
+def test_interconnect_fifo(cw_files, add_pd, io_sides, depth):
+    chip_size = 2
+    interconnect = create_cgra(chip_size, chip_size, io_sides,
+                               num_tracks=3,
+                               add_pd=add_pd,
+                               mem_ratio=(1, 2))
+
+    netlist = {
+        "e0": [("I0", "io2f_16"), ("m0", "data_in")],
+        "e4": [("i3", "io2f_1"), ("m0", "wen_in")],
+        "e5": [("i4", "io2f_1"), ("m0", "ren_in")],
+        "e1": [("m0", "data_out"), ("I1", "f2io_16")],
+        "e2": [("m0", "almost_empty"), ("i2", "f2io_1")],
+        "e3": [("m0", "almost_full"), ("i3", "f2io_1")],
+        "e6": [("m0", "valid_out"), ("i4", "f2io_1")]
+    }
+    bus = {"e0": 16, "e1": 16, "e2": 1, "e3": 1, "e4": 1, "e5": 1, "e6": 1}
+
+    placement, routing = pnr(interconnect, (netlist, bus))
+    config_data = interconnect.get_route_bitstream(routing)
+
+    # in this case we configure m0 as fifo mode
+    mode = Mode.FIFO
+    tile_en = 1
+
+    mem_x, mem_y = placement["m0"]
+    memtile = interconnect.tile_circuits[(mem_x, mem_y)]
+    mcore = memtile.core
+    config_data.append((interconnect.get_config_addr(
+                        mcore.get_reg_index("depth"),
+                        0, mem_x, mem_y), depth))
+    config_data.append((interconnect.get_config_addr(
+                        mcore.get_reg_index("mode"),
+                        0, mem_x, mem_y), mode.value))
+    config_data.append((interconnect.get_config_addr(
+                        mcore.get_reg_index("tile_en"),
+                        0, mem_x, mem_y), tile_en))
+    config_data.append((interconnect.get_config_addr(
+                        mcore.get_reg_index("almost_count"),
+                        0, mem_x, mem_y), 0))
```

Comment:
You need the `almost_count` at least to be 2 or 3

Reply:
why? if I set it to 0 my logic works such that empty == almost_empty, full == almost_full.

Reply:
Because in reality the consumers typically have some pipeline delays.

Reply:
right, but this is just a test for functionality

Reply:
fixed, used actual empty and full